crates/polars-plan/src/plans/ir/scan_sources.rs

```diff
@@ -114,26 +118,23 @@ impl ScanSources {
     ) -> PolarsResult<Self> {
         match self {
             Self::Paths(paths) => {
-                let hive_enabled = file_options.hive_options.enabled;
                 let (expanded_paths, hive_start_idx) = expand_paths_hive(
                     paths,
                     file_options.glob,
                     cloud_options,
-                    hive_enabled.unwrap_or(false),
+                    file_options.hive_options.enabled.unwrap_or(false),
                 )?;
-                let inferred_hive_enabled = hive_enabled.unwrap_or_else(|| {
-                    expanded_from_single_directory(paths, expanded_paths.as_ref())
-                });
 
-                file_options.hive_options.enabled = Some(inferred_hive_enabled);
+                if file_options.hive_options.enabled.is_none()
+                    && expanded_from_single_directory(paths, expanded_paths.as_ref())
+                {
+                    file_options.hive_options.enabled = Some(true);
```

Comment:
this does the same thing (only overwriting if the existing value is None), the old code did `hive_enabled.unwrap_or_else` which made it a bit hard to see